diff --git a/package/whetstone/whetstone.mk b/package/whetstone/whetstone.mk
index d47fbcd..c2ded13 100644
--- a/package/whetstone/whetstone.mk
+++ b/package/whetstone/whetstone.mk
@@ -9,11 +9,12 @@ WHETSTONE_SOURCE = whetstone.c
 WHETSTONE_SITE = http://www.netlib.org/benchmark/
 
 define WHETSTONE_BUILD_CMDS
-	$(TARGET_CONFIGURE_OPTS) $(MAKE) LDLIBS="-lm" -C $(@D) whetstone
+	$(TARGET_CONFIGURE_OPTS) $(MAKE) -C $(@D) results
 endef
 
 define WHETSTONE_CLEAN_CMDS
 	rm -f $(@D)/whetstone
+	rm -f $(@D)/*.o
 endef
 
 define WHETSTONE_INSTALL_TARGET_CMDS
